awscrt/http.py

```diff
@@ -0,0 +1,153 @@
+    # and is ready for use.
+    @staticmethod
+    def new_connection(bootstrap, host_name, port, socket_options, on_connection_shutdown, tls_connection_options):
+        assert tls_connection_options is None or isinstance(tls_connection_options, TlsConnectionOptions)
+        assert host_name is not None
+        assert port is not None
+        assert socket_options is not None and isinstance(socket_options, SocketOptions)
+        assert on_connection_shutdown is not None
+
+        future = Future()
+        connection = HttpClientConnection(bootstrap, on_connection_shutdown, tls_connection_options)
+
+        def on_connection_setup_native_cb(native_handle, error_code):
+            if error_code == 0:
+                connection._native_handle = native_handle
+                future.set_result(connection)
+            else:
+                future.set_exception(Exception("Error during connect: err={}".format(error_code)))
+
+        try:
+            if tls_connection_options:
+                internal_conn_options_handle = tls_connection_options._internal_tls_conn_options
+            else:
+                internal_conn_options_handle = None
+
+            _aws_crt_python.aws_py_http_client_connection_create(bootstrap._internal_bootstrap,
+                                                                 on_connection_setup_native_cb,
+                                                                 connection._on_connection_shutdown,
+                                                                 host_name,
+                                                                 port,
+                                                                 socket_options,
+                                                                 internal_conn_options_handle)
+
+        except Exception as e:
+            future.set_exception(e)
+
+        return future
+
+    # Closes the connection, if you hold a reference to this instance of HttpClientConnection, on_connection_shutdown
+    # will be invoked upon completion of the connection close.
+    def close(self):
+        if self._native_handle is not None:
+            _aws_crt_python.aws_py_http_client_connection_close(self._native_handle)
+
+    # Returns True if the connection is open and usable, False otherwise.
+    def is_open(self):
+        if self._native_handle is not None:
+            return _aws_crt_python.aws_py_http_client_connection_is_open(self._native_handle)
+
+        return False
+
+    # Makes an Http request. When the headers from the response are received, the returned future will have a result.
+    # and request.response_headers will be filled in, and request.response_code will be available.
+    # After this future completes, you can get the result of request.response_completed,
+    # for the remainder of the response.
+    def make_request(self, request):
```

Comment:
Note for future us: We should make sure this function is the same name across all languages. I think MakeRequest/make_request is probably the best, but we just need to check for this.